client/state/index.js

```diff
@@ -149,13 +149,17 @@ export function createReduxStore( initialState = {} ) {
 
 	const middlewares = [
 		thunkMiddleware,
+		// We need the data layer middleware to be used as early
+		// as possible, before any side effects.
+		// This avoids, for example, side effect actions like analytics
+		// being dispatched more than once.
```

Comment:
While this comment does go a ways to helping us understand what went wrong I think we can go a bit further to explain _why_ it has to be here…

```
// The data layer dispatches actions on network events
// including success, failure, and progress updates
// Its way of issuing these is to wrap the originating action
// with special meta and dispatch it again each time.
// If another middleware jumps in before the data layer
// then it could mistakenly trigger on those network
// responses. Therefore we need to inject the data layer
// as early as possible into the middleware chain.
```

Something like that is a bit verbose but also explains the process by which the side effects could be reissued. Take it or leave it; I think some clarity on how this happens is needed here beyond just stating _that_ it happens.